connectors/config.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+import copy
+
+from envyaml import EnvYAML
+
+
+class ConfigNotLoadedError(Exception):
+    pass
+
+
+class Config:
```

Comment:
If you want to create a singleton, it's best to use the same pattern that logging, see https://github.com/elastic/connectors-python/blob/main/connectors/logger.py

```
_CONFIG = None

def set_config(config_file):
    global _CONFIG
    _CONFIG = Config(config_file)

def get_config():
    return _CONFIG

    
```